### PR TITLE
DROOLS-1504 Resolution of variable with leading or trailing spaces

### PR DESCRIPTION
**Pull request by tarilabs**

### CODE REVIEW

File: kie-dmn/kie-dmn-api/src/main/java/org/kie/dmn/api/core/DMNMessageType.java [hunk 1]
```diff
@@ -5,7 +5,7 @@
     REQ_NOT_FOUND( "The referenced node was not found", Tag.COMPILATION, Tag.DMN_CORE ),
     TYPE_REF_NOT_FOUND( "The listed type reference could not be resolved", Tag.COMPILATION, Tag.DMN_CORE ),
     TYPE_DEF_NOT_FOUND( "The listed type definition was not found", Tag.COMPILATION, Tag.DMN_CORE ),
-    INVALID_NAME( "The listed name is not a valid FEEL identifier", Tag.COMPILATION, Tag.DMN_CORE ),
+    INVALID_NAME( "The listed name is not a valid FEEL identifier", Tag.VALIDATION, Tag.DMN_VALIDATOR, Tag.COMPILATION, Tag.DMN_CORE ),
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-api/src/main/java/org/kie/dmn/api/core/DMNMessageType.java:8]: This type is therefore also used during dmn-validation

File: kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/DMNRuntimeTest.java [hunk 2]
```diff
@@ -1018,6 +1018,30 @@ public void testWrongConstraintsInItemDefinition() {
...
+        person.put("Name", "John");
+        person.put("Surname", "Doe");
+        context.set("Input Person", person);
+
+        DMNResult dmnResult = runtime.evaluateAll( dmnModel, context );
+        
+        assertThat( formatMessages( dmnResult.getMessages() ), dmnResult.hasErrors(), is( false ) );
+        DMNContext result = dmnResult.getContext();
+        assertThat( result.get( "Further Decision" ), is( "The person was greeted with: 'Ciao John Doe'" ) );
+    }
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/DMNRuntimeTest.java:1044]: Test result context:

```
{
    Input Person: {
        Surname: Doe
        Name: John
    }
     Greet Person: Ciao John Doe
    Further Decision: The person was greeted with: 'Ciao John Doe'
}
```

File: kie-dmn/kie-dmn-validation/src/test/java/org/kie/dmn/validation/ValidatorTest.java [hunk 3]
```diff
@@ -428,6 +428,14 @@ public void testTYPEREF_NO_FEEL_TYPE() {
         assertTrue( validate.stream().anyMatch( p -> p.getMessageType().equals( DMNMessageType.TYPE_REF_NOT_FOUND ) ) );
     }
     
+    @Test
+    public void testVARIABLE_LEADING_TRAILING_SPACES() {
+        List<DMNMessage> validate = validator.validate( getReader( "VARIABLE_LEADING_TRAILING_SPACES.dmn" ), VALIDATE_SCHEMA, VALIDATE_MODEL, VALIDATE_COMPILATION);
+        assertThat( formatMessages( validate ), validate.size(), is( 1 ) );
+        assertTrue( validate.stream().anyMatch( p -> p.getMessageType().equals( DMNMessageType.INVALID_NAME ) ) );
+        assertThat( validate.get(0).getSourceId(), is("_dd662d27-7896-42cb-9d14-bd74203bdbec") );
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-validation/src/test/java/org/kie/dmn/validation/ValidatorTest.java:436]: dmn-validation new message:

```
DMNMessage{ severity=WARN, type=INVALID_NAME, message='Variable name contains leading or traling spaces ' Greet Person' on node ' Greet Person'', sourceId='_dd662d27-7896-42cb-9d14-bd74203bdbec', exception='', feelEvent=''}
```

with WARN level as requested

**Review by tarilabs — COMMENTED** [review]
@etirelli can you review please?

**Comment by tarilabs** [comment]
```
Test Result (1 failure / +1)
org.drools.compiler.test.PositionalTest.testPositionalWithNull
```

Jenkins test again :8ball:

**Comment by etirelli** [comment]
Excellent @tarilabs, +1. 

@mariofusco can you merge this please?

**Review by etirelli — APPROVED** [review]

**Pull request merged**